XMLtoMD/GuidelineXmlToMD/Program.cs

```diff
@@ -12,25 +12,64 @@ namespace GuidelineXmlToMD
     class Program
     {
         static MarkdownOut.MdWriter _MdWriter;
-        static void Main(string[] args)
+
+        /// <summary>
+        /// A simple to to convert coding guidelines xml to the website formatted markdown.
+        /// </summary>
+        /// <param name="args">It is implied that args[0] is the name of the xml file that is stored at CodingGuidelines/docs/ This can also be explicitly passed with --xml-file-name</param>
+        /// <param name="xmlFileName">The name of the xml file that is stored at CodingGuidelines/docs/</param>
+        /// <param name="i">The xml file to process when using the -m option</param>
+        /// <param name="o">The md file to create when using the -m option </param>
+        /// <param name="m">Manual Run - Use xmlFilePath and outputMDFilePath</param>
+        static void Main(string[] args, string xmlFileName, string i = "null", string o = "null", bool m = false)
         {
-            string xmlFileName = "Guidelines(8th Edition).xml";
-            if (args.Length != 0) {  //check for input fileName
-                if (Regex.Match(args[0], @".*.xml").Success) {
-                    xmlFileName = args[0];
+            string markDownOutputFilePath;
+            string xmlInputFilePath;
+
+            if (m)
+            {
+                if (Regex.Match(i, @".*.xml").Success)
```

Comment:
Slight issue with the regex with the second period matching any character. This might be cleaner as an .EndsWith call.
```suggestion
                if (i.EndsWith(".xml"))
```